parsec/core/backend_connection/authenticated.py

```diff
@@ -257,30 +277,30 @@ async def set_status(
     def get_organization_config(self) -> OrganizationConfig:
         return self._organization_config
 
-    def register_monitor(self, monitor_cb) -> None:
+    def register_monitor(self, monitor_cb: Callable[..., Any]) -> None:
         if self._started:
             raise RuntimeError("Cannot register monitor once started !")
         self._monitors_cbs.append(monitor_cb)
 
-    def are_monitors_idle(self):
+    def are_monitors_idle(self) -> bool:
         return self._monitors_idle_event.is_set()
 
-    async def wait_idle_monitors(self):
+    async def wait_idle_monitors(self) -> None:
         await self._monitors_idle_event.wait()
 
     @asynccontextmanager
-    async def run(self):
+    async def run(self) -> AsyncGenerator[None, None]:
         if self._started:
             raise RuntimeError("Already started")
-        async with trio.open_service_nursery() as nursery:
+        async with trio.open_service_nursery() as nursery:  # type: ignore[attr-defined]
```

Comment:
Good catch, could you import `open_service_nursery` from `parsec.utils` instead?